spec/integration/udf_spec.lua

```diff
@@ -0,0 +1,105 @@
+require("busted.runner")()
+local driver = require("luasql.exasol")
+local config = require("config")
+local log = require("remotelog")
+local amalg = require("amalg_util")
+
+config.configure_logging()
+local connection_params = config.get_connection_params()
+
+-- [itest -> dsn~use-available-exasol-udf-libraries-only~1]
+describe("Exasol driver works inside an UDF", function()
+    local env = nil
+    local conn = nil
+
+    before_each(function()
+        env = driver.exasol()
+        conn = env:connect(connection_params.source_name, connection_params.user, connection_params.password)
+    end)
+
+    after_each(function()
+        assert.is_true(conn:close())
+        conn = nil
+        assert.is_true(env:close())
+        env = nil
+    end)
+
+    local function create_schema()
+        local schema_name = string.format("CONNECTION_TEST_%d", os.time())
+        assert(conn:execute(string.format("drop schema if exists %s cascade", schema_name)))
+        assert(conn:execute(string.format("create schema %s", schema_name)))
+        finally(function()
+            assert(conn:execute(string.format("drop schema %s cascade", schema_name)))
+        end)
+        return schema_name
+    end
+
+    local function create_script_with_driver(schema_name, script_arguments, script_content)
+        local content = amalg.amalgamate_with_script(script_content)
+        local statement = string.format("CREATE LUA SCALAR SCRIPT %s.RUN_TEST(%s) RETURNS VARCHAR(2000) AS\n%s\n/",
+                                        schema_name, script_arguments, content)
+        assert(conn:execute(statement))
+    end
+
+    local function escape_string(string)
+        return string:gsub("'", "''")
+    end
+
+    local function execute_query_in_udf(query)
+        local script = [[
+local driver = require("luasql.exasol")
+
+local function print_result(cur)
+    local result = ""
+    result = string.format("%sColumn names: [%s]\n", result, table.concat(cur:getcolnames(), ", "))
+    result = string.format("%sColumn types: [%s]\n", result, table.concat(cur:getcoltypes(), ", "))
+    local index = 1
+    local row = {}
+    row = assert(cur:fetch(row, "n"))
+    while row ~= nil do
+        result = string.format("%sRow %d: [%s]\n", result, index, table.concat(row, ", "))
+        row = cur:fetch(row, "n")
+        index = index + 1
+    end
+    return result
+end
+
+function run(ctx)
+    local env = driver.exasol()
+    local conn = assert(env:connect(ctx.source_name, ctx.user_name, ctx.password))
+    local cur = assert(conn:execute(ctx.query))
+    local result = print_result(cur)
+    cur:close()
+    conn:close()
+    env:close()
+    return result
+end
+]]
+        local script_arguments =
+                "source_name VARCHAR(100), user_name VARCHAR(100), password VARCHAR(100), query VARCHAR(2000)"
+        local schema_name = create_schema()
+        create_script_with_driver(schema_name, script_arguments, script)
+        query = escape_string(query)
```

Comment:
Don't reuse variable. Just imagine it were immutable.

Reply:
done.